server/typesense/index.ts

```diff
@@ -6,14 +6,12 @@ const createTypesenseClient = () =>
   new Client({
     nodes: [
       {
-        host: env.TYPESENSE_HOST,
-        port: env.TYPESENSE_PORT,
-        ...(env.NODE_ENV === 'production'
-          ? { protocol: 'https' }
-          : { protocol: 'http' }),
+        host: env.NEXT_PUBLIC_TYPESENSE_HOST,
+        port: env.NEXT_PUBLIC_TYPESENSE_PORT,
+        protocol: 'http',
       },
     ],
-    apiKey: process.env.TYPESENSE_API_KEY!,
+    apiKey: env.NEXT_PUBLIC_TYPESENSE_API_KEY,
```

Comment:
Huh, why would you expose the Typesense API key to the client? This is a security risk.

Reply:
It just wasn't working without making it public, now that I think about it that shouldn't be required, I'll look that